Cleaning up Tests

lucidd opened the pull request:
This cleans up / simplifies the tests and addresses all the points of #43.

moio commented:
I agree, we can merge this. From a purely stylistic POV, I usually write `Collection`s "initializers" like this:

    List<String> myList = new LinkedList<String>() {{
        add("a");
        add("b");
        add("c");
    }};

This kind of breaks at least my automatic code formatter, as it would put newlines between curly braces, but looks clearer in its intention to me. Feel free to change your code accordingly or not!

lucidd commented:
@moio I agree this would look much better and more concise but it conflicts with the current checkstyle rules in multiple ways. A checkstyle compliant version would look like this:

```java
Map<String, String> kwargs = new LinkedHashMap<String, String>() { {
        put("refresh", "true");
        put("sysupgrade", "false");
    } };
```
Its still better then the current version despite the extra indentation and whitespace but it gets really awkward when nesting maps.
I'm not sure if we can configure checkstyle in a way that it accepts your version while still enforcing those rules in other places where they make sense.
Also the <> operator does not work for anonymous inner classes so we are stuck with the longer version.

renner commented:
Ok, let's merge it. I like @moio's version best while I'm not sure if I like the checkstyle compliant version better than the style in the commit. Feel free to do as you prefer. And could you please rebase against the current master, there is some conflicts after the merge of #65.

renner commented on src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java:
```diff
@@ -212,14 +241,43 @@ public void testRunRequestAsync() throws Exception {
                 .withHeader("Content-Type", equalTo("application/json"))
                 .withRequestBody(equalToJson(JSON_RUN_REQUEST)));
+
+
```
Maybe remove one of those double empty lines here.

renner commented on src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java:
```diff
@@ -335,15 +359,44 @@ public void testQueryJobResult() throws Exception {
+        verify(1, getRequestedFor(urlEqualTo("/jobs/some-job-id"))
+                .withHeader("Accept", equalTo("application/json")));
+
+
```
Same as above, maybe remove one empty line?

renner commented on src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java:
```diff
@@ -335,15 +359,44 @@ public void testQueryJobResult() throws Exception {
+                        put("old", "");
+                    }
+                });
+                put("i3status", new LinkedHashMap<String, String>(){
```
If we keep this collection initialization style, I would personally put a blank between all the `()` and the `{`. Might be only a personal preference though.

moio commented:
@lucidd, you are right. I just double checked and it seems there is no way for Checkstyle to digest "double brace initialization" at the moment.
Solutions could be:
 - patch Checkstyle! (in a totally different effort)
 - disable indentation checking. This is what we do in Spacewalk;
 - keep the very bulky Checkstyle-approved style or avoid double brace initialization altogether.

Your choice.

lucidd commented:
@renner I rebased it and removed the double new lines. I also adjusted the new tests from #65 to this patch. @moio I will leave the changes to the double brace ininitialization out of this patch and submit a new one once i tried out a few of those options. If its ok with you guys i will merge this as soon as travis is ok with it.

renner commented:
Fine with me, feel free to merge :+1:
